core/src/main/scala/org/apache/spark/executor/ExecutorURLClassLoader.scala

```diff
@@ -19,56 +19,51 @@ package org.apache.spark.executor
 
 import java.net.{URLClassLoader, URL}
 
-import org.apache.spark.util.ParentClassLoader
+/**
+ * A ClassLoader trait that changes the normal delegation scheme.
+ * Normally, it's (cache, parent, self).  GreedyClassLoader uses
+ * (cache, self, parent).  This lets this CL "hide" or "override"
+ * class defs that also exist in the parent loader.
+ */
+private[spark] trait GreedyClassLoader extends ClassLoader {
```

Comment:
Rather than making this a trait, can you just modify ChildExecutorURLClassLoader to have this logic? We don't need it anywhere else at this point, and this is the main intended goal of ChildExecutorURLClassLoader (to do greedy loading like this).